nucypher/policy/conditions/signing/base.py

```diff
@@ -0,0 +1,236 @@
+from abc import ABC
+from typing import Any, Optional, Tuple
+
+from marshmallow import (
+    ValidationError,
+    fields,
+    post_load,
+    validate,
+    validates,
+    validates_schema,
+)
+from marshmallow.validate import Range
+
+from nucypher.policy.conditions.base import Condition
+from nucypher.policy.conditions.context import (
+    resolve_any_context_variables,
+)
+from nucypher.policy.conditions.exceptions import (
+    InvalidContextVariableData,
+)
+from nucypher.policy.conditions.lingo import ConditionType, ReturnValueTest
+from nucypher.policy.conditions.utils import (
+    ConditionProviderManager,
+    camel_case_to_snake,
+    is_camel_case,
+)
+from nucypher.utilities.abi import (
+    decode_human_readable_call,
+    extract_arg_types,
+    is_valid_human_readable_signature,
+)
+
+SIGNING_CONDITION_OBJECT_CONTEXT_VAR = ":signingConditionObject"
+
+
+class SigningObjectCondition(Condition, ABC):
+    """
+    Base class for signing conditions.
+    This class is abstract and should not be instantiated directly.
+    """
+
+    class Schema(Condition.Schema):
+        signing_object_context_var = fields.Str(
+            required=True, validate=validate.Equal(SIGNING_CONDITION_OBJECT_CONTEXT_VAR)
+        )
+
+    def __init__(
+        self,
+        signing_object_context_var: str = SIGNING_CONDITION_OBJECT_CONTEXT_VAR,
+        *args,
+        **kwargs,
+    ):
+        self.signing_object_context_var = signing_object_context_var
+        super().__init__(*args, **kwargs)
+
+
+class SigningObjectAttributeCondition(SigningObjectCondition):
+    CONDITION_TYPE = ConditionType.ATTRIBUTE.value
+
+    class Schema(SigningObjectCondition.Schema):
+        condition_type = fields.Str(
+            validate=validate.Equal(ConditionType.ATTRIBUTE.value), required=True
+        )
+        attribute_name = fields.Str(required=True)
+        return_value_test = fields.Nested(ReturnValueTest.Schema(), required=True)
+
+        # maintain field declaration ordering
+        class Meta:
+            ordered = True
+
+        @post_load
+        def make(self, data, **kwargs):
+            return SigningObjectAttributeCondition(**data)
+
+    def __init__(
+        self,
+        attribute_name: str,
+        return_value_test: ReturnValueTest,
+        signing_object_context_var: str = SIGNING_CONDITION_OBJECT_CONTEXT_VAR,
+        condition_type: str = ConditionType.ATTRIBUTE.value,
+        name: Optional[str] = None,
+    ):
+        # TODO what about camel case (from library) vs snake case (python) for
+        #  attribute names (is this sufficient?)
+        #  At the moment since there will be a python object, can we assume snake case conversion always?
```

Comment:
I think this conversion to snake case is fine. We are already using snake case for :userAddress context variable, so it makes sense to me to use the same convention for attributes.